packages/@react-spectrum/slider/src/SliderBase.tsx

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {BaseSliderProps, SpectrumSliderTicksBase} from '@react-types/slider';
+import {classNames, useStyleProps} from '@react-spectrum/utils';
+import {LabelPosition, Orientation, StyleProps, ValueBase} from '@react-types/shared';
+import React, {CSSProperties, HTMLAttributes, MutableRefObject, ReactNode, useRef} from 'react';
+import {SliderState, useSliderState} from '@react-stately/slider';
+import styles from '@adobe/spectrum-css-temp/components/slider/vars.css';
+import {useLocale, useNumberFormatter} from '@react-aria/i18n';
+import {useProviderProps} from '@react-spectrum/provider';
+import {useSlider, useSliderThumb} from '@react-aria/slider';
+
+export interface SliderBaseChildArguments {
+  inputRefs: MutableRefObject<undefined>[],
+  thumbProps: HTMLAttributes<HTMLElement>[],
+  inputProps: HTMLAttributes<HTMLElement>[],
+  ticks: ReactNode,
+  state: SliderState
+}
+
+export interface SliderBaseProps extends BaseSliderProps, ValueBase<number[]>, SpectrumSliderTicksBase, StyleProps {
+  children: (SliderBaseChildArguments) => ReactNode,
+  formatOptions?: Intl.NumberFormatOptions,
+  classes?: string[] | Object,
+  style?: CSSProperties,
+  count: 1 | 2,
+  // SpectrumBarSliderBase:
```

Comment:
I assume you're not extending above because this includes something that conflicts. Maybe pull out these into a type we can reuse here?